grails-app/controllers/org/pih/warehouse/core/OrganizationController.groovy

```diff
@@ -102,9 +101,10 @@ class OrganizationController {
                         identifierService.generateOrganizationIdentifier(organizationInstance.name)
             }
 
-            if (!organizationInstance.hasErrors() && organizationInstance.save(flush: true)) {
-                flash.message = "${warehouse.message(code: 'default.updated.message', args: [warehouse.message(code: 'organization.label', default: 'Organization'), organizationInstance.id])}"
-                redirect(action: "edit", id: organizationInstance.id)
+            if (organizationInstance.validate()) {
+                Organization persistedOrganization = organizationDataService.save(organizationInstance)
```

Comment:
I think this variable is not necessary, you can do only `organizationDataService.save(organizationInstance)` and leave this redirect below as it was before. You are passing reference to the organizationInstance object, so it should contain the id from the database after saving.